File: src/main/java/no/fintlabs/exceptions/ArchiveResourceNotFoundException.java
package no.fintlabs.exceptions;

import no.fintlabs.gateway.instance.AbstractInstanceRejectedException;

public class ArchiveResourceNotFoundException extends AbstractInstanceRejectedException {
    public ArchiveResourceNotFoundException(String email) {
        super("No archive resource found for saksansvarligEpost='" + email + "'");
    }
}

File: src/main/kotlin/no/novari/flyt/egrunnerverv/gateway/exception/ArchiveResourceNotFoundException.kt
package no.novari.flyt.egrunnerverv.gateway.exception

import no.novari.flyt.egrunnerverv.gateway.infrastructure.slack.SlackAlertService
import no.novari.flyt.gateway.instance.exception.AbstractInstanceRejectedException

class ArchiveResourceNotFoundException(
    email: String,
    slackAlertService: SlackAlertService,
) : AbstractInstanceRejectedException(formatMessage(email)) {
    init {
        slackAlertService.sendMessage(formatMessage(email))
    }

    companion object {
        private fun formatMessage(email: String): String = "No archive resource found for saksansvarligEpost='$email'"
    }
}
